project/src/test/scala/ScalastyleInspectionsGeneratorTest.scala

```diff
@@ -0,0 +1,141 @@
+import com.typesafe.config.ConfigFactory
+import org.scalastyle._
+import org.scalatest._
+
+import scala.io.Source._
+import scala.meta._
+import scala.xml._
+
+class ScalastyleInspectionsGeneratorTest extends FlatSpec with Matchers {
+
+  "extractInspections" should "successfully extract the inspections" in {
+    val inspections = XML.load(fromResource("scalastyle/definition.xml").bufferedReader) \\ "checker"
+    //XML.load(this.getClass.getResourceAsStream("scalastyle/definition.xml")) \\ "checker"
+    val docs = XML.load(fromResource("scalastyle/documentation.xml").bufferedReader) \\ "check"
+    // XML.load(this.getClass.getResourceAsStream("scalastyle/documentation.xml")) \\ "check"
+    val conf = ConfigFactory.parseResources("scalastyle/config.conf")
+
+    val expected = List(
+      ScalastyleInspection(
+        "org.scalastyle.file.FileTabChecker",
+        "line.contains.tab",
+        "Line contains Tab",
+        "Check that there are no tabs in a file",
+        None,
+        Some("Some say that tabs are evil."),
+        WarningLevel,
+        List()
+      ),
+      ScalastyleInspection(
+        "org.scalastyle.file.FileLengthChecker",
+        "file.size.limit",
+        "File length",
+        "Check the number of lines in a file",
+        None,
+        Some("Files which are too long can be hard to read and understand."),
+        WarningLevel,
+        List(Param("maxFileLength", IntegerType, "1500"))
+      ),
+      ScalastyleInspection(
+        "org.scalastyle.scalariform.MultipleStringLiteralsChecker",
+        "multiple.string.literals",
+        "Multiple string literals",
+        "Checks that a string literal does not appear multiple times",
+        None,
+        Some(
+          "Code duplication makes maintenance more difficult, so it can be better to replace the multiple occurrences with a constant."
+        ),
+        WarningLevel,
+        List(Param("allowed", IntegerType, "1"), Param("ignoreRegex", StringType, "^\"\"$"))
+      ),
+      ScalastyleInspection(
+        "org.scalastyle.scalariform.ScalaDocChecker",
+        "scaladoc",
+        "Missing or badly formed ScalaDoc: {0}",
+        "Checks that the ScalaDoc on documentable members is well-formed",
+        Some(
+          """Ignore tokens is a comma separated string that may include the following : PatDefOrDcl (variables), TmplDef (classes, traits), TypeDefOrDcl (type definitions), FunDefOrDcl (functions)
+            |            Supported indentation styles are "scaladoc" (for ScalaDoc-style comments, with two spaces before the asterisk), "javadoc" (for JavaDoc-style comments, with a single space before the asterisk) or "anydoc" to support any style (any number of spaces before the asterisk). For backwards compatibility, if left empty, "anydoc" will be assumed.""".stripMargin
```

Comment:
Uhm... We should find a way to fix this bad identatio, maybe replacing all whitespaces after a new line with the empty char while extracting or transforming the inspections will be sufficient?

Or are these extra espaces also removed when compiling the generated code?

Reply:
Yes, I agree - as I said before, this will probably need some reformatting similarily to what the existing module does, but I'll come back to this separately once I can create those rules in Sonar to see what the formatting looks like in the UI.

Reply:
Ok.